src/main/java/efs/task/syntax/GuessNumberGame.java

```diff
@@ -13,10 +20,67 @@ public static void main(String[] args) {
     }
 
     public GuessNumberGame(String argument) {
-        //TODO: Implement the constructor
+        try {
+            M = Integer.parseInt(argument);
+        } catch (java.lang.NumberFormatException e) {
+            System.out.println(UsefulConstants.WRONG_ARGUMENT);
+            throw new java.lang.IllegalArgumentException();
+        }
+
+        if (M > UsefulConstants.MAX_UPPER_BOUND || M < 1) {
+            System.out.println(UsefulConstants.WRONG_ARGUMENT);
+            throw new java.lang.IllegalArgumentException();
+        }
+
+        numberOfAttempts = (int) (Math.log(M) / Math.log(2)) + 1;
+        remainingAttempts = numberOfAttempts;
+        randomNumber = (int) (Math.random() * (M - 1)) + 1;
     }
 
     public void play() {
-        //TODO: Implement the method that executes the game session
+        String inputString;
+        int inputInt;
+        boolean won = false;
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("<1," + M + ">");
+        while (remainingAttempts != 0) {
+            remainingAttempts--;
+            printAttemptsBar();
+            System.out.println(UsefulConstants.GIVE_ME);
+
+            inputString = scanner.nextLine();
+            try {
+                inputInt = Integer.parseInt(inputString);
+            } catch (java.lang.NumberFormatException e) {
+                System.out.println(UsefulConstants.NOT_A_NUMBER);
+                continue;
+            }
+            if (inputInt == randomNumber) {
+                won = true;
+                System.out.println(UsefulConstants.YES);
+                break;
+            } else if (inputInt > randomNumber) {
+                System.out.println(UsefulConstants.TO_MUCH);
+            } else {
+                System.out.println(UsefulConstants.TO_LESS);
+            }
+
+        }
+        if (won) {
+            System.out.print(UsefulConstants.CONGRATULATIONS);
+        } else {
+            System.out.print(UsefulConstants.UNFORTUNATELY);
+        }
+    }
+
+    private void printAttemptsBar() {
+        System.out.print('[');
+        for (int i = this.numberOfAttempts; i > this.remainingAttempts; i--) {
+            System.out.print('*');
+        }
+        for (int i = this.remainingAttempts; i > 0; i--) {
+            System.out.print('.');
+        }
```

Comment:
_clean code_ możnaby tutaj pomyśleć o użyciu:
https://www.geeksforgeeks.org/string-class-repeat-method-in-java-with-examples/